index.js

```diff
@@ -7,26 +7,74 @@ var qs       = require('querystring')
 var FormData = require('form-data')
 
 function NeoCities(user, pass, opts) {
-  this.user = user
-  this.pass = pass
-  this.opts = opts || {}
+  if (typeof pass == 'object' || pass == undefined) {
+    this.key = user
+  }
+  this.opts = opts || (typeof pass == 'object' ? pass : {})
+  if (!this.opts.key && !this.key) {
+    this.user = user
+    this.pass = pass
+  }
+  else {
+    this.opts.key = this.opts.key || true
+  }
   this.url = url.parse(this.opts.url || 'https://neocities.org')
+  this.siteurl = this.opts.siteurl ? url.parse(this.ops.siteurl) : null
   this.client = this.url.protocol == 'https:' ? https : http
+  if (this.opts.key && !this.key) {
+    this.get('key', null, function (res) {
+        this.key = res.api_key
+        if (typeof this.opts.key == 'function') this.opts.key(this)
+    }, user, pass)
+  }
 }
 
-NeoCities.prototype.get = function(method, args, callback) {
-  var path = '/api/'+method
-
-  if(args)
-    path += '?'+qs.stringify(args)
+NeoCities.prototype.download = function(files, callback) {
+  var i = 0
+  var that = this
+  var returning = false
+  
+  if (!this.siteurl)
+    this.info(function (resp) {
+      that.siteurl = url.parse(that.url.protocol + '//' + (resp.info.domain || resp.info.sitename + '.' + that.url.hostname))
+      dwnldFile()
+    })
+  else dwnldFile()
+  
+  function dwnldFile() {
+    if (i < files.length) {
+      if (files[i].path) {
+        var file = fs.createWriteStream(files[i].path)
+        file.on('finish', function() {file.close((dwnldFile))})
+        that.client.get(that.siteurl.href + files[i].name.replace(/^\/?/, '/'), function(data) {return data.pipe(file)})
+      }
+//            else {
```

Comment:
I don't think its good practice to include commented code, its better to refer the commented block for discussion. Any thoughts?

Reply:
I figured out what that commented block was intended to do. It was the start of some code that allows files to be downloaded to a variable to return without saving them to the system. The use case would be if the file is needed for immediate use, but not for later. If you think that it would be good for this library to have that feature, I can finish adding it. Otherwise, it can just be deleted.

Reply:
Any comment on this feature?